globus_automate_client/cli/callbacks.py

```diff
@@ -155,3 +198,29 @@ def flows_endpoint_envvar_callback(default_value: str) -> str:
     defining the target Flow endpoint.
     """
     return os.getenv("GLOBUS_AUTOMATE_FLOWS_ENDPOINT", default_value)
+
+
+def input_validator_callback(body: str) -> str:
+    """
+    Checks if input is a file and loads it, otherwise
+    returns the body string passed in
+    """
+    # Callbacks are run regardless of whether an option was explicitly set.
+    # Handle the scenario where the default value for an option is empty
+    if not body:
+        return body
+
+    # Reading from a file was indicated by prepending the filename with the @
+    # symbol -- for backwards compatability check if the symbol is present and
+    # remove it
+    if body.startswith("@"):
+        body = body[1:]
+
+    body_path = pathlib.Path(body)
+    if body_path.exists() and body_path.is_file():
+        with body_path.open() as f:
+            body = f.read()
+    elif body_path.exists() and body_path.is_dir():
+        raise typer.BadParameter("Expected file, received directory")
+
+    return body
```

Comment:
Can we generalize this? I think the current implementation only supports loading JSON, but if we want to support YAML everywhere, we'd need to be able to parse input as YAML too.

Now that I think about it, I think we'd essentially want to use a single function to handle reading user input. The function could then delegate to the `yaml_validator_callback` and `json_validator_callback` to load the input.

Reply:
This was sorta my attempt at generalizing the input validation. So far our input seems to be limited to two types: string or file. I do think we could have some better error handling here for handling the None body and maybe some other cases I haven't thought of. Something to think about I guess